.github/workflows/release_ci.yml

```diff
@@ -0,0 +1,106 @@
+name: release_ci
+
+on:
+  push: # Runs on push to main branch

```

Comment:
instead of running this on all push and schedule, it's probably better to run only when a new release is specified:
```
on:
  release:
    types: [published]
```
this would also allow for changes to the version number during dev without accidentally creating a release.
this would also minimize action and minute usages